### PR TITLE
Mvc overview

### PR DESCRIPTION
**Pull request by TomArcher**

### CODE REVIEW

**Comment by TomArcher** [comment]
@danroth27 @TomArcher Please review this first draft

File: mvc/overview.rst [hunk 1]
```diff
@@ -1,8 +1,68 @@
...
 
-.. include:: /../common/stub-notice.txt
+- :doc:`Models </models/index>` represent data that the user is working with in your application. For example, in an inventory application, one of the model objects might be a Product object that is tasked with retrieving information from a database, operating on that information, and then writing the updated information back to the database. In small applications, the model is often a conceptual separation instead of a physical one. For example, if the application only reads a dataset and sends it to the view, the application does not have a physical model layer and associated classes. Instead, the controller might work directly with the dataset, which takes on the role of the model.
 
-.. _issue: https://github.com/aspnet/Docs/issues/113
+- :doc:`Views </views/index>` are responsible for displaying the application's user interface (UI). Typically, this UI is created from the model data. An example would be an edit view of a Products table that displays text boxes, drop-down lists, and check boxes based on the current state of the Product object being represented in the view.
+
```
Comment (MikeWasson) [inline mvc/overview.rst]: That's true enough but in a Web API scenario, there is no UI.

File: mvc/overview.rst [hunk 2]
```diff
@@ -1,8 +1,68 @@
...
+- :doc:`Controllers </controllers/index>` process incoming requests, work with the model, and ultimately select a view that displays the user interface. In an MVC application, the view only displays information; the controller handles and responds to user input and interaction. For example, the controller handles query-string values, and passes these values to the model, which in turn might use these values to query the database.
```
Comment (MikeWasson) [inline mvc/overview.rst]: Need to distinguish client-side and server-side interactions here - MVC doesn't handle anything that happens on the client side (e.g. through JS), only the HTTP request.

Not sure what "handles query-string values" means in this context.

File: mvc/overview.rst [hunk 3]
```diff
@@ -1,8 +1,68 @@
...
+
+The MVC pattern helps you create applications that separate the different aspects of the application (input logic, business logic, and UI logic), while providing a loose coupling between these elements. The pattern specifies where each kind of logic should be located in the application:
+
+- UI logic belongs in the view
+- Input logic belongs in the controller
+- Business logic belongs in the model
```
Comment (MikeWasson) [inline mvc/overview.rst]: I think that's controversial as stated, and maybe needs to distinguish between view models and domain models.

File: mvc/overview.rst [hunk 4]
```diff
@@ -1,8 +1,68 @@
...
+- An extensible and pluggable framework. The components of the ASP.NET MVC framework are designed so that they can be easily replaced or customized. You can plug in your own view engine, URL routing policy, action-method parameter serialization, and other components. The ASP.NET MVC framework also supports the use of Dependency Injection (DI) and Inversion of Control (IoC) container models. DI enables you to inject objects into a class, instead of relying on the class to create the object itself. IOC specifies that if an object requires another object, the first objects should get the second object from an outside source such as a configuration file. This makes testing easier.
```
Comment (MikeWasson) [inline mvc/overview.rst]: This isn't really an accurate definition of IoC (see the [Wikipedia definition](https://github.com/aspnet/Docs/pull/440) which is pretty good) - also it's confusing IoC with IoC _containers_, Maybe just say DI is built in and you can use your choice of IoC containers?

File: mvc/overview.rst [hunk 5]
```diff
@@ -1,8 +1,68 @@
...
+- Support for using the markup in existing ASP.NET page (.aspx files), user control (.ascx files), and master page (.master files) markup files as view templates. You can use existing ASP.NET features with the ASP.NET MVC framework, such as nested master pages, in-line expressions (<%= %>), declarative server controls, templates, data-binding, localization, and so on.
+- Support for existing ASP.NET features. ASP.NET MVC lets you use features such as forms authentication and Windows authentication, URL authorization, membership and roles, output and data caching, session and profile state management, health monitoring, the configuration system, and the provider architecture.
+- `Content negotiation <http://www.asp.net/web-api/overview/formats-and-model-binding/content-negotiation>`_ is a mechanism defined in the HTTP specification (RFC 2616) as "the process of selecting the best representation for a given response when there are multiple representations available."
```
Comment (MikeWasson) [inline mvc/overview.rst]: Give an example? (e.g., returning XML or JSON based on the client request). Also, this link goes to Web API 2 docs.

**Pull request closed without merging**